src/ui/qt/qt_ui.h

```diff
@@ -12,6 +12,8 @@ class UiWindowQt : public UiWindow
 
     virtual void drawPoint( const Point2d & point, const PaintColor & color );
     virtual void drawLine( const Point2d & start, const Point2d & end, const PaintColor & color );
+    virtual void drawEllipse(const Point2d & center, const int xDiameter, const int yDiameter, const PaintColor & color);
+    virtual void drawRectangle(const Point2d & start, const int width, const int height, const PaintColor & color);
```

Comment:
I recommend to use 4 `Point2d` points instead.

Reply:
Is this for the rectangle only, or for ellipse as well? I think it is easier to understand drawEllipse() if parameters follow geometric convention. What do you think?

Reply:
It's only for Rectangle.

Reply:
I think we could just add a separate function that accepts 4 `Point2d`, which could draw other quadrilaterals aside from rectangle. We could call it `drawQuadrangle()` or `drawQuadrilateral` or `drawTetragon()`. `drawRectangle()` would be inaccurate name if we accept 4 `Point2d`. What do you think?

Reply:
Having only one point is not enough as user doesn't know which corner of rectangle it represents. Additionally, there is no restriction top (or bottom) line of rectangle must be parallel to X axis (in other words rectangle must no be rotated).
Let's make like this to avoid misunderstanding:
```cpp
virtual void drawRectangle(const Point2d & topLeftCorner, double width, double height, const PaintColor & color);
```
`Point2d` is **double** based structure.

Reply:
Alright! :+1: I will implement the changes after I finished reformatting my computer.